src/pybamm/config.py

```diff
@@ -0,0 +1,138 @@
+import uuid
+import os
+import platformdirs
+from pathlib import Path
+import pybamm
+import select
+import sys
+
+
+def is_running_tests():  # pragma: no cover
+    """
+    Detect if the code is being run as part of a test suite or building docs with Sphinx.
+
+    Returns:
+        bool: True if running tests or building docs, False otherwise.
+    """
+    import sys
+
+    # Check if pytest or unittest is running
+    if any(
+        test_module in sys.modules for test_module in ["pytest", "unittest", "nose"]
+    ):
+        return True
+
+    # Check for GitHub Actions environment variable
+    if "GITHUB_ACTIONS" in os.environ:
+        return True
+
+    # Check for other common CI environment variables
+    ci_env_vars = ["CI", "TRAVIS", "CIRCLECI", "JENKINS_URL", "GITLAB_CI"]
+    if any(var in os.environ for var in ci_env_vars):
+        return True
+
+    # Check for common test runner names in command-line arguments
+    test_runners = ["pytest", "unittest", "nose", "trial", "nox", "tox"]
+    if any(runner in sys.argv[0].lower() for runner in test_runners):
+        return True
```

Comment:
This should check across the entire list and not just the first index – this would work for commands like `pytest -m unit`, but not for cases where one runs tools through `nox`, or `uvx pytest`/`pipx pytest`, and similar.